replica/table.cc

```diff
@@ -597,7 +597,7 @@ const storage_group_map& storage_group_manager::storage_groups() const {
 }
 
 future<> storage_group_manager::stop_storage_groups() noexcept {
-    return parallel_for_each(_storage_groups | boost::adaptors::map_values, std::mem_fn(&storage_group::stop));
+    return parallel_for_each(_storage_groups | boost::adaptors::map_values, [] (auto sg) { return sg->stop("table removal"); });
```

Comment:
This copies sg. Maybe it's some kind of shared_ptr (can't tell due to auto), but better to avoid.

Nicer (subjective): std::bind_back(&storage_group::stop, "table removal")

Reply:
Ah, it's for sure a smart pointer, since ->.

Reply:
> This copies sg. Maybe it's some kind of shared_ptr (can't tell due to auto), but better to avoid.
> 
> Nicer (subjective): std::bind_back(&storage_group::stop, "table removal")

This does not seem to be supported by clang yet